services/ui_backend_service/api/ws.py

```diff
@@ -108,18 +108,28 @@ async def event_handler(self, operation: str, resources: List[str], data: Dict,
                 'resources': resources,
                 'data': _data
             })
-            for subscription in self.subscriptions:
+            for subscription in self.subscriptions():
                 try:
                     if subscription.disconnected_ts and time.time() - subscription.disconnected_ts > WS_QUEUE_TTL_SECONDS:
-                        await self.unsubscribe_from(subscription.ws, subscription.uuid)
+                        # We can assume that all websockets (not just this UUID) are disconnected, don't filter by UUID as well.
+                        await self.unsubscribe_from(subscription.ws)
                     else:
                         await self._event_subscription(subscription, operation, resources, _data)
                 except ConnectionResetError:
                     self.logger.debug("Trying to broadcast to a stale subscription. Unsubscribing")
-                    await self.unsubscribe_from(subscription.ws, subscription.uuid)
+                    await self.unsubscribe_from(subscription.ws)
                 except Exception:
                     self.logger.exception("Broadcasting to subscription failed")
 
+    def subscriptions(self):
+        # Grab all of the keys upfront and use that to iterate so that callers can
+        # safely modify the subscriptions dict while we are iterating through it.
+        # This is primarily useful when calling `unsubscribe_from` during the cleanup
+        # loop in the event handler.
+        for k in list(self._subscriptions.keys()):
+            for sub in self._subscriptions[k]:
```

Comment:
note to self: no KeyError possible due to self._subscriptions being a `defaultdict(list)`